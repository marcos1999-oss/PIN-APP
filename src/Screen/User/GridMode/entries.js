export const ENTRIES2 = [
  {
    id: 1,
    title: "Beautiful and dramatic Antelope Canyon",
    subtitle: "Jumbo Deal $2.99 Get 2 tacos, Jumbo jack,  48 char",
    illustration: "https://i.imgur.com/KZsmUi2l.jpg",
    // illustration: require('../../../../assets/img/deal-1.png')
  },
  {
    id: 2,
    title: "Earlier this morning, NYC",
    subtitle: "Jumbo Deal $2.99 Get 2 tacos, Jumbo jack,  48 char",
    illustration: "https://i.imgur.com/KZsmUi2l.jpg",
    // illustration: require('../../../../assets/img/deal-2.png')
  },
  {
    id: 3,
    title: "White Pocket Sunset",
    subtitle: "Lorem ipsum dolor sit amet et nuncat ",
    illustration: "https://i.imgur.com/KZsmUi2l.jpg",
    // illustration: require('../../../../assets/img/deal-3.png')
  },
  {
    id: 4,
    title: "Acrocorinth, Greece",
    subtitle: "Lorem ipsum dolor sit amet et nuncat mergitur",
    illustration: "https://i.imgur.com/KZsmUi2l.jpg",
  },
  {
    id: 5,
    title: "The lone tree, majestic landscape of New Zealand",
    subtitle: "Lorem ipsum dolor sit amet",
    illustration: "https://i.imgur.com/2nCt3Sbl.jpg",
  },
  {
    id: 6,
    title: "Middle Earth, Germany",
    subtitle: "Lorem ipsum dolor sit amet",
    illustration: "https://i.imgur.com/lceHsT6l.jpg",
  },
  {
    id: 7,
    title: "Middle Earth, Germany",
    subtitle: "Lorem ipsum dolor sit amet",
    illustration: "https://i.imgur.com/lceHsT6l.jpg",
  },
  {
    id: 8,
    title: "Middle Earth, Germany",
    subtitle: "Lorem ipsum dolor sit amet",
    illustration: "https://i.imgur.com/lceHsT6l.jpg",
  },
  {
    id: 9,
    title: "Middle Earth, Germany",
    subtitle: "Lorem ipsum dolor sit amet",
    illustration: "https://i.imgur.com/lceHsT6l.jpg",
  },
  {
    id: 10,
    title: "Middle Earth, Germany",
    subtitle: "Lorem ipsum dolor sit amet",
    illustration: "https://i.imgur.com/lceHsT6l.jpg",
  },
  {
    id: 11,
    title: "Middle Earth, Germany",
    subtitle: "Lorem ipsum dolor sit amet",
    illustration: "https://i.imgur.com/lceHsT6l.jpg",
  },
];
