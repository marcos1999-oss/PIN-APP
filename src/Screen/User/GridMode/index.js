import React from "react";
import {
  FlatList,
  RefreshControl,
  View,
  TouchableOpacity,
  Animated,
  Easing,
  Dimensions,
} from "react-native";
import { Item, Input, Content } from "native-base";
import numeral from 'numeral'
import styles from "./style";
import VIPCard from "../../../Components/VIPCard";
import DealCard from "../../../Components/DealCard";
import NoDealsView from "../NoDeals";
//Redux
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import {
  getPostsPagination,
} from '../../../redux/selectors/index';
import * as postActions from '../../../redux/actions/postActions';
import {
  verticalScale,
} from "../../../Utils/scaling";

//GridView

class GridMode extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      searchValue: '',
      refreshing: false,
      isEnded: false,
    };
  }

  onPressDeal = (item) => {
    this.props.navigation.navigate("UserAdView", { isUser: true });
  };

  _keyExtractor = (item, index) => item.id.toString();

  distance = (destLat, destLong) => {
    if (this.props.me !== undefined && destLat !== undefined && destLong !== undefined) {
      const lat1 = this.props.me.latitude;
      const lon1 = this.props.me.longitude;
      const lat2 = destLat;
      const lon2 = destLong;
      const R = 6371; // km (change this constant to get miles)
      const dLat = (lat2 - lat1) * Math.PI / 180;
      const dLon = (lon2 - lon1) * Math.PI / 180;
      const a = Math.sin(dLat / 2) * Math.sin(dLat / 2) +
        Math.cos(lat1 * Math.PI / 180) * Math.cos(lat2 * Math.PI / 180) *
        Math.sin(dLon / 2) * Math.sin(dLon / 2);
      const c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1 - a));
      const d = (R * c) / 1.60934;
      if (d < 1) return Math.round(d.toFixed(2) * 10) / 10;
      else return Math.round(d);
    }
  }

  _renderItem = ({ item, index }) => {
    const distance = this.distance(item.firm.lat, item.firm.lng);
    return (
      <CustomItem
        index={index + 1}
        style={[
          styles.cardContainer,
          (index + 1) % 2 == 0
            ? styles.evenCardContainer
            : styles.oddCardContainer,
        ]}
        onPress={({ item }) => this.onPressDeal({ item })}
      >
        {item.id == 9 ? (
          <VIPCard />
        ) : (
            <DealCard
              image={item.photo.url}
              title={item.title}
              viewsCount={numeral(item.viewsCount).format('0,0')}
              likesCount={numeral(item.likesCount).format('0,0')}
              distance={distance}
            />
          )}
      </CustomItem>
    );
  };

  _onRefresh = () => {
    this.setState({ refreshing: true });
    // console.log("is refreshing");
    setTimeout(() => {
      this.setState({ refreshing: false });
    }, 5000);
  };

  _onChangeText = (searchValue) => {
    this.setState({ searchValue }, () => {
      this.props.onSearch(this.state.searchValue);
    }
    );
  };

  renderEmpty = () => {
    return (<NoDealsView />)
  };

  handlePagination() {
    if (this.props.pagination.currentPage !== this.props.pagination.total) {
      this.props.postActions.fetchPosts({
        currentPage: this.props.pagination.currentPage + 1,
        perPage: this.props.pagination.perPage,
      });
    }
  }

  render() {
    if (this.state.isEnded) {
      return this.renderEmpty()
    } else if (this.props.posts.length === 0 && !this.props.hasSearch) {
      return this.renderEmpty()
    }
    return (
      <View style={styles.container}>
        {this.props.hasSearch && (
          <Item regular style={styles.searchContainer}>
            <Input
              placeholder="Find what you looking forâ€¦"
              style={styles.searchInput}
              value={this.state.searchValue}
              onChangeText={this._onChangeText}
            />
          </Item>
        )}
        <View style={{ flex: 1 }}>
          <FlatList
            data={this.props.posts}
            numColumns={2}
            showsVerticalScrollIndicator={false}
            renderItem={this._renderItem.bind(this)}
            keyExtractor={(item, idx) => item.id}
            contentContainerStyle={styles.dealsContainer}
            onEndReachedThreshold={0.2}
            onEndReached={() => this.handlePagination()}
            getItemLayout={(data, index) => (
              { length: verticalScale(247.67), offset: verticalScale(247.67) * index, index }
            )}
            refreshControl={
              <RefreshControl
                refreshing={this.state.refreshing}
                onRefresh={this._onRefresh.bind(this)}
                title="refreshing ..."
              />
            }
          />
        </View>
      </View>
    );
  }
}

export class CustomItem extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      easeAnimated: new Animated.Value(0),
    };
  }

  componentDidMount() {
    // this.easeAnimate.setValue(0);
    Animated.timing(this.state.easeAnimated, {
      toValue: 1,
      duration: 1000,
      delay: this.props.index * 500,
      easing: Easing.out(Easing.quad),
      useNativeDriver: true,
    }).start();
  }

  render() {
    const transform = [
      {
        translateY: this.state.easeAnimated.interpolate({
          inputRange: [0, 1],
          outputRange: [styles.cardContainer.height * 2, 0],
        }),
      },
    ];
    return (
      <TouchableOpacity onPress={this.props.onPress}>
        <Animated.View style={[this.props.style, { transform }]}>
          {this.props.children}
        </Animated.View>
      </TouchableOpacity>
    );
  }
}

const mapStateToProps = state => ({
  pagination: getPostsPagination(state),
});

const mapDispatchToProps = dispatch => ({
  postActions: bindActionCreators(postActions, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(GridMode)
