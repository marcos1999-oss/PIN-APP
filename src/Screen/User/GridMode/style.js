import React from "react";
import { StyleSheet, Platform } from "react-native";
import {
  verticalScale,
  scale,
  normalizeSize,
  moderateScale,
} from "../../../Utils/scaling";

const IS_IOS = Platform.OS === "ios";
export const endedIconSize = verticalScale(100);

export default StyleSheet.create({
  endedContainer: {
    flex: 1,
    alignItems: "center",
    flexDirection: "column",
    paddingTop: verticalScale(120),
  },
  endedText: {
    marginTop: verticalScale(20),
    fontFamily: "Helvetica",
    fontSize: moderateScale(14),
    textAlign: "center",
  },
  container: {
    paddingTop: verticalScale(20),
    paddingHorizontal: scale(15),
    flex: 1,
  },
  cardContainer: {
    width: scale(167.6),
    height: verticalScale(247.67),
    marginBottom: verticalScale(13),
    backgroundColor: "white",
    shadowColor: "#000",
    shadowOffset: { width: 0, height: verticalScale(3) },
    shadowOpacity: 0.16,
    shadowRadius: 6,
    elevation: IS_IOS ? 1 : 5,
  },
  evenCardContainer: {
    marginLeft: verticalScale(5),
  },
  oddCardContainer: {
    marginRight: verticalScale(5),
  },
  dealsContainer: {
    marginBottom: verticalScale(13),
  },
  searchContainer: {
    alignSelf: "center",
    height: verticalScale(40),
    width: scale(311),
    marginHorizontal: scale(15),
    marginBottom: verticalScale(30),
    borderRadius: verticalScale(10),
    borderWidth: 0,
    borderColor: "white",
    backgroundColor: "white",
    shadowColor: "#000",
    shadowOffset: { width: 0, height: verticalScale(3) },
    shadowOpacity: 0.16,
    // shadowRadius: 6,
    elevation: IS_IOS ? 1 : 9,
  },
  searchInput: {
    fontSize: moderateScale(12),
    color: "black",
  },
});
