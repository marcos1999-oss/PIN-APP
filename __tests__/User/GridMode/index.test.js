/**
 * @format
 * @lint-ignore-every XPLATJSCOPYRIGHT1
 */

import 'react-native';
import React from 'react';
import configureMockStore from "redux-mock-store";
import mockComponent from 'react-native/jest/mockComponent';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import GridMode from '../../../src/Screen/User/GridMode';

const mockStore = configureMockStore();

const store = mockStore({
  posts: {}
});

jest.mock('react-native-cached-images', () => {
  return {
    CustomCachedImage: mockComponent('Image'),
    CachedImage: mockComponent('Image'),
    DocumentDir: () => {},
    ImageCache: {
      get: {
        clear: () => {},
      },
    },
  }
});

test('renders correctly', () => {
  const posts = [{
    id: '01',
    title: 'demo test',
    firm: {
      lat: 60.155,
      lng: -1.145,
    },
    photo: {
      url: 'https://www.easydrawingtips.com/wp-content/uploads/2019/10/spiky_male_hair_drawing.jpg',
    },
    viewsCount: 5,
    likesCount: 5,
  }]
  const postsMeta = {
    loading: true,
  };
  const tree = renderer.create(<GridMode posts={posts} postsMeta={postsMeta} store={store} />).toJSON();
  expect(tree).toMatchSnapshot();
});
