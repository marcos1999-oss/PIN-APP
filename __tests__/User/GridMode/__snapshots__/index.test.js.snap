// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingHorizontal": 30,
      "paddingTop": 40,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Object {
          "marginBottom": 26,
        }
      }
      data={
        Array [
          Object {
            "firm": Object {
              "lat": 60.155,
              "lng": -1.145,
            },
            "id": "01",
            "likesCount": 5,
            "photo": Object {
              "url": "https://www.easydrawingtips.com/wp-content/uploads/2019/10/spiky_male_hair_drawing.jpg",
            },
            "title": "demo test",
            "viewsCount": 5,
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      getItemLayout={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={2}
      onContentSizeChange={[Function]}
      onEndReached={[Function]}
      onEndReachedThreshold={0.2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      refreshControl={
        <RefreshControlMock
          onRefresh={[Function]}
          refreshing={false}
          title="refreshing ..."
        />
      }
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <RCTRefreshControl />
      <View>
        <View
          style={null}
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "white",
                    "elevation": 1,
                    "height": 495.34,
                    "marginBottom": 26,
                    "marginRight": 10,
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 6,
                      "width": 0,
                    },
                    "shadowOpacity": 0.16,
                    "shadowRadius": 6,
                    "transform": Array [
                      Object {
                        "translateY": 990.68,
                      },
                    ],
                    "width": 335.2,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {},
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "white",
                        "flex": 1,
                        "flexDirection": "column",
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {},
                        Object {
                          "height": 409.34,
                          "width": "100%",
                        },
                      ]
                    }
                  >
                    <Image
                      source={
                        Object {
                          "uri": "https://www.easydrawingtips.com/wp-content/uploads/2019/10/spiky_male_hair_drawing.jpg",
                        }
                      }
                      style={
                        Object {
                          "flex": 1,
                          "resizeMode": "cover",
                        }
                      }
                    />
                  </View>
                  <View
                    style={
                      Array [
                        Object {},
                        Object {
                          "height": 86,
                          "marginTop": 16,
                          "paddingHorizontal": 8,
                          "width": "100%",
                        },
                      ]
                    }
                  >
                    <Text
                      numberOfLines={1}
                      style={
                        Array [
                          Object {
                            "color": "#000",
                            "fontFamily": "System",
                            "fontSize": 16,
                          },
                          Object {
                            "color": "#6E6969",
                          },
                          Object {
                            "alignSelf": "flex-start",
                            "fontSize": 18,
                          },
                        ]
                      }
                      uppercase={false}
                    >
                      demo test
                    </Text>
                    <View
                      style={
                        Array [
                          Object {},
                          Object {
                            "flexDirection": "row",
                            "marginTop": 16,
                          },
                        ]
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {},
                            Object {
                              "alignItems": "flex-start",
                              "flexDirection": "row",
                              "marginRight": 20,
                            },
                          ]
                        }
                      >
                        <Text
                          allowFontScaling={false}
                          style={
                            Array [
                              Object {
                                "color": undefined,
                                "fontSize": 12,
                              },
                              Object {
                                "color": "#DF0103",
                                "fontSize": 12,
                                "marginTop": 4,
                              },
                              Object {
                                "fontFamily": "PinAppIcon",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                              Object {},
                            ]
                          }
                        >
                          юдк
                        </Text>
                        <Text
                          style={
                            Array [
                              Object {
                                "color": "#000",
                                "fontFamily": "System",
                                "fontSize": 16,
                              },
                              Object {
                                "color": "#6E6969",
                                "fontSize": 12,
                                "marginLeft": 10,
                              },
                            ]
                          }
                          uppercase={false}
                        >
                          5
                           Likes
                        </Text>
                      </View>
                      <View
                        style={
                          Array [
                            Object {},
                            Object {
                              "alignItems": "flex-start",
                              "flexDirection": "row",
                            },
                          ]
                        }
                      >
                        <Text
                          allowFontScaling={false}
                          style={
                            Array [
                              Object {
                                "color": undefined,
                                "fontSize": 12,
                              },
                              Object {
                                "color": "#000",
                                "fontSize": 12,
                                "marginTop": 4,
                              },
                              Object {
                                "fontFamily": "PinAppIcon",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                              Object {},
                            ]
                          }
                        >
                          юдй
                        </Text>
                        <Text
                          style={
                            Array [
                              Object {
                                "color": "#000",
                                "fontFamily": "System",
                                "fontSize": 16,
                              },
                              Object {
                                "color": "#6E6969",
                                "fontSize": 12,
                                "marginLeft": 10,
                              },
                            ]
                          }
                          uppercase={false}
                        >
                          5
                           Views 
                        </Text>
                      </View>
                      <Text
                        style={
                          Array [
                            Object {
                              "color": "#000",
                              "fontFamily": "System",
                              "fontSize": 16,
                            },
                            Object {
                              "color": "#9F9C9C",
                              "fontSize": 10,
                              "fontWeight": "bold",
                              "marginLeft": 10,
                            },
                          ]
                        }
                        uppercase={false}
                      >
                         mi
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
